test/images.js

```diff
@@ -50,84 +47,85 @@ describe('Images', function() {
 
   this.timeout(t.timeout);
 
-  after(function(done) {
-    return t.destroy(apos, done);
+  after(function() {
+    return t.destroy(apos);
   });
 
-  it('should be a property of the apos object', function(done) {
+  it('should be a property of the apos object', async function() {
     this.timeout(t.timeout);
     this.slow(2000);
 
-    apos = require('../index.js')({
+    apos = await require('../index.js')({
       root: module,
       shortName: 'test',
+      argv: {
+        _: []
+      },
       modules: {
         'apostrophe-express': {
-          port: 7900
+          port: 7900,
+          address: 'localhost',
+          session: {
+            secret: 'Porta'
+          }
         }
-      },
-      afterInit: function(callback) {
-        assert(apos.images);
-        // In tests this will be the name of the test file,
-        // so override that in order to get apostrophe to
-        // listen normally and not try to run a task. -Tom
-        apos.argv._ = [];
-        return callback(null);
-      },
-      afterListen: function(err) {
-        assert(!err);
-        done();
       }
     });
+
+    assert(apos.images);
+    assert(apos.images.__meta.name === 'apostrophe-images');
   });
 
   // Test pieces.list()
-  it('should clean up any existing images for testing', function(done) {
-    // Newer mongo returns a promise from remove even if there's a callback,
-    // which in turn confuses mocha if we use a return statement here. So don't. -Tom
-    apos.docs.db.remove({ type: 'apostrophe-image' }, function(err) {
-      assert(!err);
-      done();
-    });
+  it('should clean up any existing images for testing', async function() {
+    try {
+      const response = await apos.docs.db.remove({ type: 'apostrophe-image' }
+      );
+      assert(response.result.ok === 1);
+    } catch (e) {
+      assert(false);
+    }
   });
 
-  it('should add images for testing', function(done) {
+  it('should add images for testing', async function() {
     assert(apos.images.insert);
-    return async.each(mockImages, function(image, callback) {
-      return apos.images.insert(apos.tasks.getReq(), image, callback);
-    }, function(err) {
-      assert(!err);
-      done();
+
+    const req = apos.tasks.getReq();
+
+    const insertPromises = mockImages.map(async (image) => {
```

Comment:
This is a very faithful translation! I am tempted to suggest simplifying it to a `for...of` loop with an `await` inside, but your way we continue to know it can handle parallelism.